tests/test_consumer.py

```diff
@@ -103,3 +107,62 @@ def test_json_row_batch_pickle_out_of_band() -> None:
     buffers: MutableSequence[PickleBuffer] = []
     data = pickle.dumps(batch, protocol=5, buffer_callback=buffers.append)
     assert pickle.loads(data, buffers=[b.raw() for b in buffers]) == batch
+
+
+def get_row_count(storage: Storage) -> int:
+    return (
+        storage.get_cluster()
+        .get_query_connection(ClickhouseClientSettings.INSERT)
+        .execute(f"SELECT count() FROM {storage.get_schema().get_local_table_name()}")[
+            0
+        ][0]
+    )
+
+
+def test_multistorage_strategy() -> None:
+    from snuba.datasets.storages import groupassignees, groupedmessages
+
+    from tests.datasets.cdc.test_groupassignee import TestGroupassignee
+    from tests.datasets.cdc.test_groupedmessage import TestGroupedMessage
+
+    commit = Mock()
+
+    storages = [groupassignees.storage, groupedmessages.storage]
+
+    strategy = MultistorageConsumerProcessingStrategyFactory(
+        storages, 10, 10, 1, int(32 * 1e6), int(64 * 1e6), TestingMetricsBackend(),
+    ).create(commit)
+
+    payloads = [
+        KafkaPayload(None, b"{}", [("table", b"ignored")]),
+        KafkaPayload(
+            None,
+            json.dumps(TestGroupassignee.INSERT_MSG).encode("utf8"),
+            [("table", groupassignees.storage.get_postgres_table().encode("utf8"))],
+        ),
+        KafkaPayload(
+            None,
+            json.dumps(TestGroupedMessage.INSERT_MSG).encode("utf8"),
+            [("table", groupedmessages.storage.get_postgres_table().encode("utf8"))],
+        ),
+    ]
+
+    messages = [
+        Message(
+            Partition(Topic("topic"), 0), offset, payload, datetime.now(), offset + 1
+        )
+        for offset, payload in enumerate(payloads)
+    ]
+
+    with assert_changes(
+        lambda: get_row_count(groupassignees.storage), 0, 1
+    ), assert_changes(lambda: get_row_count(groupedmessages.storage), 0, 1):
+
+        for message in messages:
+            strategy.submit(message)
+
+        with assert_changes(
+            lambda: commit.call_args_list, [], [call({Partition(Topic("topic"), 0): 3})]
+        ):
+            strategy.close()
+            strategy.join()
```

Comment:
May I ask additional details about this so we can improve the Storage api if it makes test hard. Till this PR the consumer was not depending on the registry as it was passed the storage directly and the only reason you had to add that dependency was because the processor is not pickable so you had to reload everything.
Or are you referring to a different problem ?

(The content of the registry can be faked for testing, though I am not sure what you were aiming at so I am not sure that will help you).

Reply:
(Going to follow up with this but submitting first set of replies as I've already written a lot.)